tock/utilization/views.py

```diff
@@ -50,3 +57,92 @@ def get_context_data(self, **kwargs):
         context.update({'org_totals': org_billing_context()}
         )
         return context
+
+
+def _plot_utilization(dates, billable, nonbillable):
+    """Make a stacked area plot of billable and nonbillable hours.
+
+    dates, billable, and nonbillable should be sequences with the same length
+    """
+    fig = go.Figure()
+
+    fig.add_trace(go.Scatter(
+        x=dates,
+        y=billable,
+        line_shape="hv",
+        mode="lines+markers",
+        stackgroup="one",
+        name="billable",
+    ))
+    fig.add_trace(go.Scatter(
+        x=dates,
+        y=nonbillable,
+        line_shape="hv",
+        mode="lines+markers",
+        stackgroup="one",
+        name="nonbillable",
+    ))
+
+    fig.update_layout(
+        # autosize=False,
+        # width=900,
+        # height=500,
+        xaxis=dict(autorange=True),
+        yaxis=dict(autorange=True),
+        xaxis_title="Reporting Period Start Date",
+        yaxis_title="Hours",
+        title="Total Hours recorded vs. Time",
+    )
+
+    plot_div = plot(fig, output_type='div', include_plotlyjs=False)
+    return plot_div
+
+
+def _utilization_data(start_date, end_date):
+    Timecard = apps.get_model("hours", "Timecard")
```

Comment:
We're[ storing sums for billable and non-billable on each Timecard](https://github.com/18F/tock/blob/main/tock/hours/models.py#L247) can we leverage those here instead of aggregating across all `timecardobjects`?

Reply:
I don't see any timecards in the test database that have non-null values for `billable_hours` and `non_billable_hours`. 

Is that a problem with the `test_data/`? Given that the migration for those fields comes from 2020-04-03, it seems likely that the test database just hasn't been updated to match the presence of the new fields...which makes it hard to test the implementation of this feature.

I'd rather not block this on waiting for #986 to be done first, but that would be a possible way forward.

Reply:
@neilmb wrangled up some test data and I believe has a WIP commit to refactor this a bit. Resolving for now.